File: 백준/Silver/7481. ATM놀이/ATM놀이.java
import java.io.*;
import java.util.StringTokenizer;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        while (T > 0) {
            T -= 1;
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int s = Integer.parseInt(st.nextToken());

            int max = Math.max(a, b);
            int min = Math.min(a, b);

            int limit = s / max;
            boolean[] flag = new boolean[min];
            boolean possible = false;

            for (int i = limit; i >= 0; i--) {
                int remain = s - i * max;
                int mod = remain % min;

                if (flag[mod]) {
                    break;
                }
                flag[mod] = true;
                if (mod == 0) {
                    possible = true;
                    if (max == a) {
                        sb.append(i + " " + remain / min + "\n");
                    } else {
                        sb.append(remain / min + " " + i + "\n");
                    }
                }
            }

            if (!possible) {
                sb.append("Impossible\n");
            }
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/7481. ATM놀이/ATM놀이.py
T = int(input())
for _ in range(T):
    possible = False
    a, b, S = map(int, input().split())
    max_v = max(a, b)
    min_v = min(a, b)

    limit = S // max_v

    flag = [False] * min_v
    check = False
    for i in range(limit, -1, -1):
        remain = S - max_v * i
        mod = remain % min_v
        if flag[mod]:
            break

        flag[mod] = True
        if mod == 0:
            check = True
            answer = []
            if max_v == a:
                answer.append(i)
                answer.append(remain // min_v)
            else:
                answer.append(remain // min_v)
                answer.append(i)
            break
            
    if check:
        print(*answer)
    else:
        print("Impossible")